lib/react.dart

```diff
@@ -769,7 +780,7 @@ abstract class Component2 implements Component {
   /// > [Consider recommended alternative solutions first!](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#preferred-solutions)
   ///
   /// See: <https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops>
-  Map getDerivedStateFromProps(Map nextProps, Map prevState) => null;
+  Map? getDerivedStateFromProps(Map nextProps, Map prevState) => null;
```

Comment:
Is it possible for `prevState` to be null on first mount? Or is this lifecycle method not called then?

Reply:
Good question, it looks like it isn't possible for it to be null. 

On first mount, `prevState` is the value of `initialState`

This behavior is validated here in tests: https://github.com/Workiva/react-dart/blob/ae23239d2ff798baa5ff4a13a4682760d6ed1db5/test/lifecycle_test.dart#L1005

Since intialState is non-nullable and defaults to an empty map, and the underlying component's `state` is always initialized upon mount:
https://github.com/Workiva/react-dart/blob/0616cd087f5433c117f09ceb83fcecf19e6665a1/lib/src/react_client/dart_interop_statics.dart#L219
then it will always be non-null.

This also lines up with TypeScript's typing, showing this arg as the same type as the state generic parameter `S` (and not `S | undefined`): https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f800de4ffd291820a9e444e6b6cd3ac9b4a16e53/types/react/index.d.ts#L710